Dockerfile

```diff
@@ -0,0 +1,16 @@
+FROM python:3.9 as builder
+
+ADD https://mirror.openshift.com/pub/openshift-v4/x86_64/clients/oc/latest/linux/oc.tar.gz /tmp/oc.tar.gz
+RUN cd /tmp; tar -xvf oc.tar.gz
+
+RUN pip install poetry
```

Comment:
Any specific reason why poetry is used?

Reply:
I use poetry because:

- It doesn't ignore `pip.conf` so you have a single place to configure alternate package repositories
- It manages builds and publishing as well
- It works with `pyproject.toml`

Reply:
> I use poetry because:
> 
> * It doesn't ignore `pip.conf` so you have a single place to configure alternate package repositories

Is there any intention to use an alternative package repository? This might be interesting for us to recommend/consume packages in Thoth.

Reply:
My use case is that I run a PyPi mirror on my local system to alleviate PyPi performance problems (and provide a "search" endpoint, since PyPi's has been disabled).

Reply:
> [...] on my local system to [...]

This should really not be an argument for a design decision ;)

Reply:
> I think it should be. I think the features that tooling affords the developer are a _critical_ part of the design decision.

Was Pipenv limiting the configuration in some way? I think [it provides a way to configure package repositories in one place](https://pipenv-fork.readthedocs.io/en/latest/advanced.html#specifying-package-indexes) that is project-specific and does not collide with the system setup.

> * It manages builds and publishing as well

If you mean publishing the built artifacts, that is something that [twine](https://pypi.org/project/twine/) does as well and is the official PyPA tool for that. Nevertheless, we have an [automated pipeline](https://github.com/AICoE/aicoe-ci) that should free developers of this task.

> * It works with `pyproject.toml`

Is there any value added by working with `pyproject.toml`?

Reply:
Unless there's a specific technical objection you'd like to raise here, I'm not interesting in revisiting this decision.